ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,199 @@
+            int[] tr2 = { 5, 6, 6, 7 };
+            int[] tr3 = { 1, 1, 1, 1 };
+            int[] tr4 = { 6, 10, 0, 1, 0, 1, 6 };
+            int[] tr5 = { 4, 5, 6, 7 };
+            int unique;
+            bool found = FindUnique(tr1, out unique);
+            Console.WriteLine($"Unique test 1: {unique}");
+            found = FindUnique(tr2, out unique);
+            Console.WriteLine($"Unique test 2: {unique}");
+            found = FindUnique(tr3, out unique);
+            Console.WriteLine($"Unique test 3: {unique}");
+            found = FindUnique(tr4, out unique);
+            Console.WriteLine($"Unique test 4: {unique}");
+            found = FindUnique(tr5, out unique);
+            Console.WriteLine($"Unique test 5: {unique}");
+
+        }
+
+        public static void PrintArray(int[] array)
+        {
+            foreach (int i in array)
+            {
+                Console.Write($"{i} ");
+            }
+            Console.WriteLine();
+        }
+        public static void Randomise(int[] array, int min, int max)
+        {
+            Random random = new Random();
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = random.Next(min, max);
+            }
+        }
+
+        public static bool FindUnique(int[] array, out int unique)
+        {
+            bool[] repeaters = new bool[array.Length];
+            unique = -1;
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                unique = array[i];
+                for (int j = 0; j < array.Length - 1; j++)
+                {
+                    if (j != i && unique == array[j])
+                    {
+                        repeaters[j] = true;
+                        repeaters[i] = true;
+                    }
+                }
+            }
+            for (int i = 0; i < repeaters.Length; i++)
+            {
+                if (!repeaters[i])
+                {
+                    unique = array[i];
+                    return true;
+                }
+            }
+            unique = 47;
+            return false;
+        }
+
+        public static void Sort(int[] array, SortAlgorithmType type, OrderBy order)
+        {
+            Console.WriteLine("Array before soring: ");
+            PrintArray(array);
+            switch (type)
+            {
+                case SortAlgorithmType.Insertion:
+                    Console.WriteLine("Insertion sort");
+                    InsertionSort(array);
+                    break;
+                case SortAlgorithmType.Selection:
+                    Console.WriteLine("Selection sort");
+                    SelectionSort(array);
+                    break;
+                case SortAlgorithmType.Bubble:
+                    Console.WriteLine("Bubble sort");
+                    BubbleSort(array);
+                    break;
+                default:
+                    break;
+            }
+            if (order == OrderBy.Descending)
```

Comment:
Descending means sorting from highest to lowest, so you need to make a decision on how to sort inside your sort methods